### DISCUSSION TITLE
Coding on Android Studio, some dependencies fail to resolve

### DISCUSSION THREAD

**Discussion by Lehua18** [community/community#138389] [Programming Help]
### Body

Hi everyone!
My class is coding an app in android studio koala, using kotlin. I am getting the following errors:

- Failed to resolve: io.github.jan-tennert.supabase:bom:2.6.1
- Failed to resolve: io.ktor:ktor-client-[engine]:0.9.5
- Failed to resolve: io.ktor:ktor-client-cio:0.9.5
- Failed to resolve: io.supabase:supabase-android:0.1.0
- Failed to resolve: io.supabase:supabase-kt:1.24.07
- Failed to resolve: io.github.jan-tennert.supabase:core-kt
- Failed to resolve: io.github.jan-tennert.supabase:auth-kt
- Failed to resolve: io.github.jan-tennert:supabase:Tag
- Failed to resolve: io.github.jan-tennert.supabase:supabase-android:2.6.1
- Failed to resolve: io.github.jan-tennert.supabase:supabase-realtime:2.6.1
- Failed to resolve: io.github.jan-tennert.supabase:runtime:2.6.1

Here is a gradle scan if it helps: https://gradle.com/s/l2afx6gsj4oi6/activate/dt3cqtrq3iwxk

Thanks!

UPDATE:
When I try to run the project, I also get the following errors:

- > Task :app:checkDebugAarMetadata FAILED
- > Task :app:mapDebugSourceSetPaths FAILED
- > Task :app:mergeDebugResources FAILED
- > Task :app:processDebugMainManifest FAILED
- > Task :app:mergeDebugAssets FAILED
- > Task :app:checkDebugDuplicateClasses FAILED
- > Task :app:desugarDebugFileDependencies FAILED
- > Task :app:mergeDebugNativeLibs FAILED

It also appears that part of the error is related to classpath.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by igorpereirag** [comment]
It looks like you're encountering dependency resolution issues in your Android project. Here are some steps to troubleshoot and potentially fix the errors:

### 1. **Check Your Gradle Configuration:**
Ensure you have the correct repositories added to your `build.gradle` files, both in the project-level and app-level `build.gradle`. Dependencies like Supabase and Ktor may require specific repositories.

Add these to your **project-level `build.gradle`**:

```groovy
allprojects {
    repositories {
        google()
        mavenCentral()
        jcenter() // Sometimes older dependencies are found here
    }
}
```

Also, ensure you have the latest version of Gradle by updating the **Gradle wrapper**:

```groovy
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
```

### 2. **Verify Dependency Versions:**
Some of the versions you are using seem outdated or incorrect. For instance, Ktor `0.9.5` is very old. The current stable version of Ktor is 2.x. You should update the Ktor dependencies to the latest versions.

For **Ktor**:
```kotlin
implementation("io.ktor:ktor-client-core:2.0.0")
implementation("io.ktor:ktor-client-cio:2.0.0")
```

For **Supabase**:
Make sure you are using the correct coordinates for Supabase Kotlin SDK. Check the official Supabase Kotlin GitHub repository for the latest versions. It looks like you're using outdated or incorrect group names.

```kotlin
implementation("io.github.jan-tennert.supabase:supabase-kt:2.6.1")
```

### 3. **Correct Supabase Dependencies:**
You might want to confirm the exact artifact IDs for Supabase. They might have been renamed or updated. Visit [Supabase Kotlin SDK](https://github.com/supabase-community/supabase-kt) for the correct versions and artifacts.

### 4. **Invalidate Caches and Restart Android Studio:**
Sometimes, Android Studio can have caching issues with dependencies. To fix this:
- Go to **File > Invalidate Caches / Restart > Invalidate and Restart**.

### 5. **Update Gradle Sync:**
After making these changes, force a Gradle sync by clicking on the "Sync Now" button that appears at the top of Android Studio, or go to **File > Sync Project with Gradle Files**.

### 6. **Ensure Internet Connection and Firewall:**
Sometimes the issue could be with network restrictions or firewalls blocking Gradle from downloading dependencies.

---

**Reply by Lehua18** [reply]
I've tried most of these things. The following are my issues with each:

1. I get an error every time I add 
`allprojects {
    repositories {
        google()
        mavenCentral()
        jcenter() // Sometimes older dependencies are found here
    }
}` 
to `build.gradle` I get the following error message: Build was configured to prefer settings repositories over project repositories but repository 'Google' was added by build file 'app\build.gradle.kts'

2. I updated my ktor version, which did not fix the errors. I still get the same error with the updated version: Failed to resolve: io.ktor:ktor-client-[engine]:2.0.0. Can someone link the Supabase Kotlin GitHub repository so I can look in it?


3. 3-6 have all been done are should be correct.

Thanks for the ideas, though!

**Reply by gnusco** [reply]
Hi,
I guess I'm esperiencing the same problem as Lehua, 
working environment is AndroidStudio & java: 

in build.gradle.kts I have:

        implementation (platform(libs.supabase.bom))
        implementation (libs.io.github.jan.tennert.supabase.supabase.kt2)
        implementation ("io.github.jan-tennert.supabase:storage-kt")

sync is ok, also Invalidate Caches and Restart Android Studio has been done,
in spite of this, the result is that import doesn't work:

        import io.github.jan_tennert.supabase.SupabaseClient;
        import io.github.jan_tennert.supabase.createSupabaseClient;
        import io.github.jan_tennert.supabase.storage.StorageClient;

"cannot resolve the symbol jan_tennert".
any advice?
Thanks

**Reply by The-lYNCAN** [reply]
facing the exact same problem as the guys/gals above. have tried literally every point that igorpereirag wrote but the error still persists

**Reply by The-lYNCAN** [reply]
a seventh point which I have already tried is the minimum sdk version of the app should be 26, initially I was creating an app with a minimum sdk version less than the required but I tried changing it to 27 and the errors still persists.

**Reply by The-lYNCAN** [reply]
I think there might be a problem with the sdk. next I am gonna try creating a new project and see if it works on that. I am praying to god that it doesn't work in the new project as well
